test/test_positioning.rb

```diff
@@ -47,6 +48,30 @@ def test_no_duplicate_row_values
 
     list.destroy
   end
+
+  def test_no_duplicate_row_values_when_advisory_lock_is_disabled_and_parent_record_is_locked
+    ActiveRecord::Base.connection_handler.clear_all_connections!
+
+    list = List.create name: "List"
+    items = []
+
+    10.times do
+      threads = 20.times.map do
+        Thread.new do
+          ActiveRecord::Base.connection_pool.with_connection do
+            list.with_lock do
```

Comment:
I think my original test doesn't test updates either? I think have a go at it here and see how you get on and I might back-fill the other one too.